jobs/ServiceInstanceUpdateJob.js

```diff
@@ -39,25 +40,23 @@ class ServiceInstanceUpdateJob extends BaseJob {
         logger.error(msg);
         return this.runFailed(new errors.ServiceUnavailable(msg), operationResponse, job, done);
       }
-      return this
-        .getServicePlanIdForInstanceId(instanceDetails.instance_id)
-        .then((planId) => (planId === undefined ?
+      return eventmesh.apiServerClient.getResource({
```

Comment:
Resource group and types are all plan related. Instead wouldnt it be better if we hit the info end point of the service instance by servicefabrik api controller and consume that? Because the instance could also be K8S down the line and this code is typing up the code more to director. So i would recommend enhancing FabrikAPI to return instance info if required (if it is already not returning it). -- i.e. in case if agents are down of if agents cannot be down if the info endpoint is currently erroring, instead of erroinng they can include the state and return back other info from etcd as part of that api.

Reply:
As no provisioner is present apart from bosh director, hence hard coded. Ideal would be to save plan id in job data. In plan the resource group and type are present.
So, have put comment
```
      /* In case of instance provisioned by other than bosh director.
       * Need to chnage resource group / type. Better to save plan name in job data.
       * Otherwise need to look for instance_id with all possible provisioners.
       */
```